### PR TITLE
Add last_update_git_commit_hash field to Branch

### PR DESCRIPTION
**Pull request by seankimdev**
Add `last_update_git_commit_hash` string field to the `RepositoryBranch`. 
The field will be used for the purpose of rendering the associated git commit of the branch's last updated event. 

It may be an empty  if the branch's last update does not have any git commits associated.

### CODE REVIEW

**Review by unmultimedio — APPROVED** [review]
Lgtm

File: proto/buf/alpha/registry/v1alpha1/repository_branch.proto [hunk 1]
```diff
@@ -33,9 +33,12 @@ message RepositoryBranch {
   google.protobuf.Timestamp last_update_time = 5;
   // The ID of the user who updated the branch.
   string last_update_user_id = 6;
-  // The author name of the most recent associated git commit of the branch. May be null
+  // The author name of the most recent associated git commit of the branch. May be an empty string
   // if the last branch update doesn't have a git commit associated.
   string last_update_git_author_name = 7;
+  // The git commit hash of the most recent associated git commit of the branch. May be an empty string
+  // if the last branch update doesn't have a git commit associated.
+  string last_update_git_commit_hash = 8;
```
Comment (bufdev) [inline proto/buf/alpha/registry/v1alpha1/repository_branch.proto:41]: Do we use the terminology `hash` to refer to git commits anywhere else in our APIs?

Comment (seankimdev) [reply proto/buf/alpha/registry/v1alpha1/repository_branch.proto:41]: Yes, it's used in several places, including [GitCommitInformation](https://buf.build/bufbuild/buf/docs/main:buf.alpha.registry.v1alpha1#buf.alpha.registry.v1alpha1.GitCommitInformation) and [GitSyncPoint](https://buf.build/bufbuild/buf/docs/main:buf.alpha.registry.v1alpha1#buf.alpha.registry.v1alpha1.GitSyncPoint).
Do you see this being an issue?

Comment (bufdev) [reply proto/buf/alpha/registry/v1alpha1/repository_branch.proto:41]: OK

**Pull request merged**